src/marine/godae/trak2ioda.py

```diff
@@ -0,0 +1,206 @@
+        '''
+
+        # idata is the dictionary containing IODA friendly data structure
+        idata = self.odata
+
+        # Diff. between ob. time and time in the file (hrs)!
+        for i in range(self.odata['n_obs']):
+            dtg = datetime.strptime(self.odata['ob_dtg'][i], '%Y%m%d%H%M')
+            idata['ob_dtg'][i] = (dtg - self.date).seconds/3600.
+
+        idata['nlocs'] = len(idata['ob_dtg'])
+
+        self.idata = idata
+
+        return
+
+    def to_ioda(self):
+
+        if self.odata['n_obs'] <= 0:
+            print('No trak observations for IODA!')
+            return
+
+        self._to_nlocs()
+
+        ncid = netcdf.netcdf_file(self.filename + '.nc', mode='w')
+
+        ncid.createDimension('nlocs', self.idata['nlocs'])  # no. of obs per var type
+        ncid.createDimension('nrecs', 1)  # no. of track obs
+        ncid.createDimension('nvars', 1)  # no. of variables
+        ncid.createDimension('nobs', 1*self.idata['nlocs'])  # total no. of obs
+
+        longitudes = ncid.createVariable('longitude@MetaData', 'f4', ('nlocs',))
+        longitudes.units = 'degrees_east'
+
+        latitudes = ncid.createVariable('latitude@MetaData', 'f4', ('nlocs',))
+        latitudes.units = 'degrees_north'
+
+        times = ncid.createVariable('time@MetaData', 'f4', ('nlocs',))
+        times.long_name = 'observation time from date_time'
+        times.units = 'hours'
+        times.description = 'why does this have to be with a reference to, why not absolute?'
+
+        sal = ncid.createVariable('sea_surface_salinity@ObsValue', 'f4', ('nlocs',))
+        sal.units = 'g/kg'
+        sal.description = ''
+
+        sal_errs = ncid.createVariable(
+            'sea_surface_salinity@ObsError', 'f4', ('nlocs',))
+        sal_errs.units = '(g/kg)^2'
+        sal_errs.description = 'Standard deviation of observation error'
+
+        sst = ncid.createVariable(
+            'sea_surface_temperature@ObsValue', 'f4', ('nlocs',))
+        sst.units = 'degree_celcius'
+        sst.description = ''
+
+        sst_errs = ncid.createVariable(
+            'sea_surface_temperature@ObsError', 'f4', ('nlocs',))
+        sst_errs.units = '(degree_celcius)^2'
+        sst_errs.description = 'Standard deviation of observation error'
+
+        u = ncid.createVariable(
+            'sea_surface_zonal_wind@ObsValue', 'f4', ('nlocs',))
+        u.units = 'm/s'
+        u.description = ''
+
+        u_errs = ncid.createVariable(
+            'sea_surface_zonal_wind@ObsError', 'f4', ('nlocs',))
+        u_errs.units = '(m/s)^2'
+        u_errs.description = 'Standard deviation of observation error'
+
+        v = ncid.createVariable(
+            'sea_surface_meridional_wind@ObsValue', 'f4', ('nlocs',))
+        v.units = 'm/s'
+        v.description = ''
+
+        v_errs = ncid.createVariable(
+            'sea_surface_meridional_wind@ObsError', 'f4', ('nlocs',))
+        v_errs.units = '(m/s)^2'
```

Comment:
same as above, std deviation instead of variance.

Reply:
I simply followed what is in profile2ioda.py.
I also noticed that in profile2ioda.py, someone added a floor value of 0.5 to the temperature obs. err.
The ob err. in track and surface obs. is currently arbitrary and needs to be addressed at some point.
At some point we will standardize these to use the netCDF API from @srherbener for writing to IODA format.

Reply:
fixed.